.github/workflows/test.yml

```diff
@@ -42,14 +42,27 @@ jobs:
           yarn install --immutable
           yarn run build
 
-      - name: "Start web server and run tests"
+      - name: "Start web server"
         shell: bash -el {0}
         run: |
-          yarn run start & pytest --video on --output test-results --screenshots true test/playwright/test_ux.py
+          yarn run start &
+          wait-for-it localhost:8000 -t 240   # wait for conda-store-ui to start (seconds)
 
-      - name: "Upload artifacts"
-        uses: actions/upload-artifact@v3
+      - name: "Run playwright tests"
+        shell: bash -el {0}
+        run: |
+          pytest --video on --output video --screenshots true test/playwright/test_ux.py
+
+      - name: "Upload screenshot artifacts"
+        uses: actions/upload-artifact@v4
         if: ${{ always() }}
         with:
-          name: playwright-tests
+          name: screenshots
           path: test-results
+
+      - name: "Upload video artifact"
```

Comment:
I'm not quite sure I understand how these video-related changes in the workflow file relate to the rest of the code diff?

Reply:
First, I broke out the starting of the web server into a separate stage from the playwright tests. IIRC this had something to do with the `wait-for-it`, but it also helps with visibility/breakdown as its running. 

The video stuff was added for two reasons: 
1. Everything was passing locally for me, but failing on CI and I couldn't "see" why. 
2. Having the video makes it easier to debug when things fail and clear what the UI looks like and what's happening. The playwright errors commonly occur much later in the code than the first failure, which can be seen on the video.
3. For someone like myself that has to grab images of the current UI for presentations and whatnot, its handy.

Reply:
Ah, okay, I was looking for specific video-related commands, but this is essentially uploading a video transcript of the entire testing session. Could you add a comment to the file explaining why you are uploading the video and screenshots? Because for someone just coming into this workflow file, it's maybe not immediately obvious why in a workflow file about testing, the workflow file is uploading screenshots and videos. It doesn't seem obviously related.

Reply:
done!